test/Sentry.Tests/Protocol/SentryEventTests.cs

```diff
@@ -80,4 +82,88 @@ public void Modules_Getter_NotNull()
         var evt = new SentryEvent();
         Assert.NotNull(evt.Modules);
     }
+
+    [Fact]
+    public void Redact_Redacts_Urls()
+    {
+        // Arrange
+        var message = "message123 https://user@not.redacted";
+        var logger = "logger123 https://user@not.redacted";
+        var platform = "platform123 https://user@not.redacted";
+        var serverName = "serverName123 https://user@not.redacted";
+        var release = "release123 https://user@not.redacted";
+        var distribution = "distribution123 https://user@not.redacted";
+        var moduleValue = "module123 https://user@not.redacted";
+        var transactionName = "transactionName123 https://user@sentry.io";
+        var requestUrl = "https://user@sentry.io";
+        var username = "bob";
+        var email = "bob@127.0.0.1";
+        var ipAddress = "127.0.0.1";
+        var environment = "environment123 https://user@not.redacted";
+
+        var breadcrumbMessage = "message https://user@sentry.io"; // should be redacted
+        var breadcrumbDataValue = "data-value https://user@sentry.io"; // should be redacted
+        var tagValue = "tag_value https://user@not.redacted";
+
+        var timestamp = DateTimeOffset.MaxValue;
+
+        var evt = new SentryEvent()
+        {
+            Message = message,
+            Logger = logger,
+            Platform = platform,
+            ServerName = serverName,
+            Release = release,
+            Distribution = distribution,
+            TransactionName = transactionName,
+            Request = new Request()
+            {
+                Method   = "GET",
+                Url = requestUrl
+            },
+            User = new User()
```

Comment:
@mattjohnsonpint should these three `User` properties also be redacted? e.g. bob@foo.com -> [Filtered]@foo.com?

Same again the in the Transaction (which also has a user)...